src/distributions.jl

```diff
@@ -37,7 +37,7 @@ function xavier(a...)
         fanout = size(w, ndims(w))
         fanin = div(length(w), fanout)
     end
-    s = convert(eltype(w), sqrt(2 / (fanin + fanout)))
+    s = convert(eltype(w), sqrt(6 / (fanin + fanout)))
```

Comment:
I think, our version is specialized for conv layers with relu activation. The part you changed is called as `gain`. You may want to update your pr to allow the xavier function to accept the gain parameter. And its default value can be 6.

Reply:
To be honest, I barely know the theoretical background. I guess you are referring to "Delving Deep into Rectifiers" paper when you say it is specialized for conv layers with relu activation. In the paper, it states this should hold: `n_l * var(w_l) = 2` where `n_l` is the average number of units per layer.  You can check that:
`x = xavier(200,300)`
`(200+300) / 2 * var(x) ~= 0.33`, where this value should be 1.0 for Xavier, 2.0 for ReLU activation. I also compared `xavier` with Tensorflow's equivalent [initializer](https://www.tensorflow.org/api_docs/python/tf/contrib/layers/variance_scaling_initializer). TF's xavier is ~3 times of `xavier`, and TF's kaiming (relu specialized xavier) is ~6 times of `xavier`, consistently.

As for your suggestion I am very new to Julia and I couldn't find a way to edit arguments so that it is compatible with pre-existing models. However there can be another distribution that takes both `gain` and `n` as arguments (as in TF).

Reply:
You can use keyword arguments for options.
```julia
xavier(a...; gain = 6)
```